src/App.js

```diff
@@ -107,134 +109,199 @@ export default class App extends React.Component<{}, AppState> {
             'forward1',
             'moveCharacter',
             () => {
-                this.audioManager.playSound('forward1');
+                // TODO: Enable announcements again.
+                // this.audioManager.playAnnouncement('forward1');
+
                 this.setState((state) => {
+                    const newCharacterState = state.characterState.forward(
+                        1,
+                        state.drawingEnabled
+                    );
+
+                    // We have to start the sound here because this is where we know the new character state.
+                    this.audioManager.playSoundForCharacterState("movement", state.settings.movementDelayMs, newCharacterState);
+
                     return {
-                        characterState: state.characterState.forward(
-                            1,
-                            state.drawingEnabled
-                        )
+                        characterState: newCharacterState
                     };
                 });
-                return Utils.makeDelayedPromise(1750);
+                return Utils.makeDelayedPromise(this.state.settings.movementDelayMs);
             }
         );
 
         this.interpreter.addCommandHandler(
             'forward2',
             'moveCharacter',
             () => {
-                this.audioManager.playSound('forward2');
+                // TODO: Enable announcements again.
+                // this.audioManager.playAnnouncement('forward2');
+
                 this.setState((state) => {
+                    const finalNewCharacterState = state.characterState.forward(2, state.drawingEnabled);
```

Comment:
In most of the other command handlers, you are using `newCharacterState`. Please update the command handlers to use a consistent name.

Reply:
Yes, that was left over from the previous approach to handling `forward` / `forward2` / `forward3`.  I standardised on `newCharacterState`, as there's no longer a need to distinguish the changes further.